components/org.wso2.carbon.identity.conditional.auth.functions.test.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/test/utils/sequence/JsSequenceHandlerRunner.java

```diff
@@ -0,0 +1,653 @@
+        } catch (IOException e) {
+            throw new JsTestException("Error in creating mock HttpServletResponse", e);
+        }
+
+        return res;
+    }
+
+    public ServiceProvider loadServiceProviderFromResource(String spFileName, Object loader) throws JsTestException {
+
+        InputStream inputStream = loader.getClass().getResourceAsStream(spFileName);
+        OMElement documentElement = null;
+        try {
+            documentElement = new StAXOMBuilder(inputStream).getDocumentElement();
+        } catch (XMLStreamException e) {
+            throw new JsTestException("Error in reading Service Provider file at : " + spFileName, e);
+        }
+        return ServiceProvider.build(documentElement);
+    }
+
+    public AuthenticationContext createAuthenticationContext(ServiceProvider serviceProvider) {
+
+        AuthenticationContext authenticationContext = new AuthenticationContext();
+        authenticationContext.setServiceProviderName(serviceProvider.getApplicationName());
+        authenticationContext.setTenantDomain("test_domain");
+        authenticationContext.setCurrentStep(1);
+        authenticationContext.setContextIdentifier(UUID.randomUUID().toString());
+        return authenticationContext;
+    }
+
+    protected void resetAuthenticators() {
+
+        FrameworkServiceDataHolder.getInstance().getAuthenticators().clear();
+        FrameworkServiceDataHolder.getInstance().getAuthenticators()
+                .add(new MockAuthenticator("BasicMockAuthenticator", new MockSubjectCallback()));
+        FrameworkServiceDataHolder.getInstance().getAuthenticators().add(new MockAuthenticator("HwkMockAuthenticator"));
+        FrameworkServiceDataHolder.getInstance().getAuthenticators().add(new MockAuthenticator("FptMockAuthenticator"));
+    }
+
+    protected static class MockSubjectCallback implements SubjectCallback, Serializable {
+
+        private static final long serialVersionUID = 597048141496121100L;
+
+        @Override
+        public AuthenticatedUser getAuthenticatedUser(AuthenticationContext context) {
+
+            AuthenticatedUser result = AuthenticatedUser.createLocalAuthenticatedUserFromSubjectIdentifier("test_user");
+            result.getUserAttributes().put(ClaimMapping
+                                                   .build("http://wso2.org/claims/givenname",
+                                                          "http://wso2.org/claims/givenname", "Test", false),
+                                           "Test");
+            result.getUserAttributes().put(ClaimMapping
+                                                   .build("http://wso2.org/claims/lastname",
+                                                          "http://wso2.org/claims/lastname", "Test", false),
+                                           "User");
+            return result;
+        }
+    }
+
+    private static class MockServletRequest implements HttpServletRequest {
```

Comment:
Wouldn't it be better to mock this using mockito as required?

Reply:
@pulasthi7 using mockito will cause dependency issues when the dependent project uses different version. Hence it was intentionally not done.